test/unit/eHealth.locations/services/selected-location-factory.js

```diff
@@ -94,39 +99,63 @@ describe('Service: SelectedLocationFactory', function () {
+      function selectFirst(depth) {
+        var level = location.levels[depth];
+        // here i am emulating user actions bound to Javascript
+        // objects via Angular, so i use this interface even if it
+        // is less convenient
+        level.selected = level.items[0];
+        level.update();
+      }
       beforeEach(function() {
         location = selectedLocationFactory();
       });
-      it('has quite a large number of third level items', function() {
-        expect(location.levels[2].items.length).toBe(850);
+
+      it('shows unfiltered options initially', function() {
+        expect(location.levels[FIRST].items.length).toBe(16)
+        expect(location.levels[SECOND].items.length).toBe(180)
+        expect(location.levels[THIRD].items.length).toBe(850);
+        expect(location.levels[FOURTH].items.length).toBe(13361);
       });
       describe('when first and second level get selected', function() {
-        function selectFirst(depth) {
-          var level = location.levels[depth];
-          // here i am emulating user actions bound to Javascript
-          // objects via Angular, so i use this interface even if it
-          // is less convenient
-          level.selected = level.items[0];
-          level.update();
-        }
         beforeEach(function() {
-          selectFirst(0);
-          selectFirst(1);
+          selectFirst(FIRST);
+          selectFirst(SECOND);
         });
         it('limits the number of options for the third level', function() {
-          expect(location.levels[2].items.length).toBe(14);
+          expect(location.levels[THIRD].items.length).toBe(14);
+        });
+        // skip test: fix not yet implemented
+        xit('does limits the number for the fourth level', function () {
+          expect(location.levels[FOURTH].items.length).toBe(13361);
         });
         describe('when third and fourth level get selected', function() {
           beforeEach(function() {
-            selectFirst(2);
-            selectFirst(3);
+            selectFirst(THIRD);
+            selectFirst(FOURTH);
           });
-          it('still show limited options for the third level', function() {
-            expect(location.levels[2].items.length).toBe(14);
+          it('still show limited options for the preceding levels', function() {
+            expect(location.levels[FIRST].items.length).toBe(16);
+            expect(location.levels[SECOND].items.length).toBe(8);
+            expect(location.levels[THIRD].items.length).toBe(14);
           });
         });
       });
+      describe('when a lower level without selecting the parent levels first', function () {
+        beforeEach(function () {
+          selectFirst(THIRD)
+        });
+        it('filters the higher level', function () {
+          expect(location.levels[SECOND].items.length).toBe(8);
+        });
+        // skip test: fix not yet implemented
+        xit('filters the current level', function () {
```

Comment:
Nope, I'm using your TDD approach :p 
Additional bugs originating in this repo were discovered, so I wrote tests for them. Tickets to fix them are being created in CC.